### PR TITLE
Slider: Add minimumRange configuration for range sliders (partially closes #22067)

### LINKED ISSUE #22067: Slider with range enabled enforces a margin
### Which Umbraco version are you using?

17.2.1

### Bug summary

When using a Slider with Range enabled a margin equal to step size is enforced.

It disregards legit use cases, e.g. age ranges, where some event may be limited only to 6 year olds but other events may be appropriate for the age of 6-10.

This was not the case in earlier versions of Umbraco (tested in 13.10.1) 

### Specifics

_No response_

### Steps to reproduce

Create a Data type with property editor `Umb.PropertyEditorUi.Slider` and the folloing configuration:

Image

Create a property on a Document type using the data type configured above and test it in the content section:

Image

It is not possible to select the same value as min and max in the slider.

### Expected result / actual result

I expect that the maximum and minimum values can be equal as in earlier versions of Umbraco.

This can be added either by allowing the margin of the slider to be configured on the Data Type or by going back to margin always being 0.

### PR DESCRIPTION
**Pull request by AndyButland**
## Description

https://github.com/umbraco/Umbraco-CMS/issues/22067 raises a specific issue about not being able to select a zero gap in a range slider, which is a legitimate use case.  More generally, we can't currently configure a "minimum gap" for a range slider, even though the underlying UUI library component supports it.

This PR adds a new `minimumRange` configuration setting on the Slider property editor that controls the minimum allowed difference between low and high values when range is enabled.

If defaults to `0`, which _should_ restore the Umbraco 13 behaviour where both handles can select the same value (e.g. age range "6 to 6").

## Known limitation: `minGap=0` not yet supported by UUI

The `uui-range-slider` component currently uses JavaScript truthiness checks (`if (this._minGap)`) to determine whether `minGap` was set. Since `0` is falsy, setting `minGap=0` is treated the same as "not configured", and the component falls back to using `step` as the minimum gap between handles.

**Impact:** With the default `minimumRange=0`, the slider UI still enforces a gap of one `step` between handles. Server-side validation correctly allows equal values, but the UI prevents selecting them.

**Upstream fix needed in [Umbraco.UI](https://github.com/umbraco/Umbraco.UI):** https://github.com/umbraco/Umbraco.UI/issues/1341

Once the UUI fix ships, `minGap=0` will work without any further changes in this codebase.

### Automated
- [x] `dotnet test --filter "FullyQualifiedName~SliderPropertyValueEditorTests"` — additional tests have been added to validate the minimum range

### Manual

1. Create a new Slider data type with: **Enable range** = true, **Min** = 0, **Max** = 10, **Step** = 1, **Minimum range** = 0.
2. Add the data type to a document type and create content.
3. Verify both handles can be dragged freely and validation passes when saving.
4. Change **Minimum range** to 3.
5. Set the slider to e.g. low=4, high=5 (gap of 1) — verify validation fails on save with a "minimum range" error.
6. Set the slider to low=4, high=7 (gap of 3) — verify validation passes.
7. Test with decimal steps: **Step** = 0.1, **Minimum range** = 0.2 — set low=0.1, high=0.3 (gap of 0.2) and verify it passes (this confirms a floating-point fix also added in this PR).

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/packages/core/components/input-slider/input-slider.element.ts [hunk 1]
```diff
@@ -40,6 +40,33 @@ function undefinedFallbackToString(value: number | undefined, fallback: number):
...
 }
 
+/**
+ * Counts the number of decimal places in a number.
+ * @param num
+ */
+function countDecimalPlaces(num: number): number {
+	const str = num.toString();
+	const decimalIndex = str.indexOf('.');
+	return decimalIndex >= 0 ? str.length - decimalIndex - 1 : 0;
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Web.UI.Client/src/packages/core/components/input-slider/input-slider.element.ts:50]: Steps this small aren't a realistic scenario, so we can avoid this complexity here.

File: src/Umbraco.Web.UI.Client/src/packages/property-editors/slider/property-editor-ui-slider.element.ts [hunk 2]
```diff
@@ -110,6 +113,7 @@ export class UmbPropertyEditorUISliderElement
 
 		this._min = this.#parseNumber(config.getValueByAlias('minVal')) || 0;
 		this._max = this.#parseNumber(config.getValueByAlias('maxVal')) || 100;
+		this._minimumRange = this.#parseNumber(config.getValueByAlias('minimumRange')) || 0;
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Web.UI.Client/src/packages/property-editors/slider/property-editor-ui-slider.element.ts]: Will address this by clamping to a minimum of zero.

**Review by kjac — COMMENTED** [review]
The server-side changes look fine from my perspective.

There is an edge case if the minimum range is configured as a negative value, which is not covered by the tests, nor handled by any validation. If it's an easy fix, that would be nice to have included. Not super important, though 😄

**Comment by AndyButland** [comment]
Thanks @kjac - I've updated to handle a negative minimum range as 0.

**Review by kjac — COMMENTED** [review]
Looks good 👍

**Review by iOvergaard — APPROVED** [review]

File: src/Umbraco.Web.UI.Client/src/packages/property-editors/slider/manifests.ts [hunk 3]
```diff
@@ -48,6 +48,14 @@ export const manifests: Array<UmbExtensionManifest> = [
 						propertyEditorUiAlias: 'Umb.PropertyEditorUi.Decimal',
 						config: [{ alias: 'step', value: '0.00001' }],
 					},
+					{
```
Comment (nielslyngsoe) [inline src/Umbraco.Web.UI.Client/src/packages/property-editors/slider/manifests.ts:51]: This is causing a conflict. It is already added in the Property Editor Schema, causing problems in 17.4. Will create a fix for it, but it surfaces that we neither have tests of creating a Silder or have tested that part on this PR.
